app/services/scholars_archive/caching_service.rb

```diff
@@ -2,21 +2,68 @@
 
 module ScholarsArchive
   class CachingService
+
     def self.fetch_or_store_in_cache(uri, expires_in)
-      Rails.cache.fetch(uri, expires_in: expires_in.to_i.hours, race_condition_ttl: 15.seconds) do
-        fetch_data(uri)
-      end
+      fetch_or_cache_json_from_etag(uri, expires_in)
     end
 
     private
+      def self.fetch_or_cache_json_from_etag(uri, expires_in)
+        etag = fetch_etag(uri)
+        data = read_etag_from_cache(uri)
+        json = nil
+        if data.nil? || data != etag
+          cache_etag(uri, etag, expires_in)
```

Comment:
rather than catching the etag here, fetch the json, grab the etag from the fetched json, cache the json, and finally cache the etag from the fetch. 

The benefit here is that you're using the etag from the request that supplied the data, and only caching the etag after a successful fetching and caching of data.